server/src/index.ts

```diff
@@ -51,9 +51,9 @@ io.on('connection', (socket: any) => {
   socket.on('ping', (ack) => {
   // The (ack) parameter stands for "acknowledgement." This function sends a message back to the originating socket.
       console.log(`[WS] Recieved ping from user <${socket.id}>.`)
-      ack('pong')
+      if (ack) ack('pong')      
   })
-  socket.on('message', (message:Message, ack) => {
+  socket.on('message', async (message, ack) => {
```

Comment:
You could technically still make this `async ( message: Message, ack)` and then go into the message types and just add a ? in front of the geohash variable (as in it could exist or not)

Reply:
Updated